activerecord/test/cases/scoping/default_scoping_test.rb

```diff
@@ -176,6 +177,24 @@ def test_unscope_string_where_clauses_involved
     assert_equal expected.sort, received.sort
   end
 
+  def test_unscope_nested_attributes_where_clause
+    david_author = Author.find_by(name: "David")
+
+    received = david_author.comments_having_post_tag_count_one.unscope(where: { posts: :tags_count })
+    expected = david_author.comments
+
+    assert_equal expected, received
+  end
+
+  def test_unscope_nested_attributes_where_clause_for_unknown_columns
+    expected = Developer.all
+    received_1 = Developer.where(name: "Jamis").unscope(where: :name)
+    received_2 = Developer.where(name: "Jamis").unscope(where: { name: :unknown_column })
```

Comment:
This PR will allows `.unscope(where: { column1: :column2, column3: :column4 })` as `.unscope(where: [:column1, :column2, :column3, :column4])`.
Is that what you want?

Reply:
@kamipo : Yes. 

I want it in that way because `unscope` is using `WhereClause#except` internally which would skip the columns. `except` here(https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation/where_clause.rb#L143) is trying to check if columns(in our case `[:column1, :column2, :column3, :column4]`) includes the `attr.name` and reject if true. 

Currently, on `master` branch, the columns value for `where: { column1: :column2, column3: :column4 }` goes like -> `["[:column1, :column2]", "[:column3, :column4]"]` due to which unscope fails for nested attributes. 

Is this the correct way to approach this problem?

Reply:
We usually regard nested hash as `{ table1: { column1: value1 }, table2: { column2: value2 } }`.
That `{ column1: :column2, column3: :column4 }` is regarded as `[:column1, :column2, :column3, :column4]` looks odd to me.

Reply:
Actually, on `master` branch -> `{ column1: :column2, column3: :column4 }` is regarded as `[[:column1, :column2], [:column3, :column4]]` if hash is passed. 

Following fail here(https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation/where_clause.rb#L143) 
```
[[:column1, :column2], [:column3, :column4]].include?(attr.name.to_s)
``` 
that is the reason we have flatten the columns to `[:column1, :column2, :column3, :column4]`

The values are not considered here so I think it fine to flatten. Let me know your  thoughts on it.

Reply:
The flatten makes a passed hash no longer behaves no-op.
That is considered as a breaking change that silently leaking information when the existing apps are upgraded to newer Rails.

Reply:
Thanks, @kamipo . I didn't know about `flatten` leaking information. I'll try to fix it some other way. 

Any thoughts on how I should approach the problem?